interfaces/ldap/v0/schema.py

```diff
@@ -0,0 +1,59 @@
+"""This file defines the schemas for the provider and requirer sides of this
+relation interface.
+
+It must expose two interfaces.schema_base.DataBagSchema subclasses called:
+- ProviderSchema
+- RequirerSchema
+"""
+
+from typing import Optional
+
+from interface_tester.schema_base import DataBagSchema
+from pydantic import AnyUrl, BaseModel, Field
+
+
+class LdapProviderData(BaseModel):
+    ldap_url: AnyUrl = Field(
+        description="The LDAP URL",
+        example="ldap://ldap.canonical.com",
+    )
+    base_dn: str = Field(
+        description="The base entry as the starting point for LDAP search "
+                    "operation",
+        example="dc=canonical,dc=com",
+    )
+    bind_dn: str = Field(
+        description="The DN of the bind account",
+        example="cn=admin,ou=engineering,dc=canonical,dc=com",
+    )
+    bind_password: str = Field(
+        description="The juju secret URI of the bind account's password",
+        example="secret://59060ecc-0495-4a80-8006-5f1fc13fd783/cjqub6vubg2s77p3nio0"
+    )
+    auth_method: str = Field(
+        description="The LDAP authentication method",
+        example="simple",
+    )
+    starttls_enabled: bool = Field(
+        description="The indicator of StartTLS operation enabled or not",
+        example=True,
+    )
+
+
+class LdapRequirerData(BaseModel):
+    user: Optional[str] = Field(
+        description="The user name provided by the requirer charmed operator",
+    )
+    group: Optional[str] = Field(
+        description="The group name provided by the requirer charmed operator",
+    )
```

Comment:
The values provided for `user` and `group` are totally up to the requirer. The `user` and `group` are for the provider to identify the requirer charmed operator because there will be a bind account created for each requirer.

In terms of the valid character restrictions, it does exist in LDAP ([some resources](https://www.ibm.com/docs/en/sva/10.0.6?topic=tasks-valid-characters-ldap-user-group-names)). Try to avoid using those restricted characters. If there is a reason that one has to be used, we can escape it.

I will add some explanation to the doc.